### DISCUSSION TITLE
Where are my brackets ?

### DISCUSSION THREAD

**Discussion by Dalhte** [NBISweden/AGAT#364] [General]
Hello there
I'm trying to make a custom GTF by merging a standard GTF with a handmade one. The handmade one which contains reannotated transcript, lakes level1 info that are present in the standard GTF. I have merge my two GFT files (using either the cat function of AGAT merge) and than sort the merge GTF using agat_convert_sp_gxf2gxf.pl. However, when I want to use it in cellranger-arc to creat a genome file, I got: "Property 'transcript_id' not found in GTF line XXXX" enven if this line do contains the transcript-id. But I realized that there is a difference between pre processed and post processed GTF by AGAT: the bracket arount the ID have vanished (eg: transcript_id=ENSRNOT00000093111 instead of transcript_id="ENSRNOT00000093111"). Is there a way to get them back, I beleive cellranger-arc love them ?
Best
David

**Comment by Juke34** [comment]
By default the agat_convert_sp_gxf2gxf.pl output is GFF3. So you have GFF file. You need an extra step to run the GFF to GTF conversion  and it should be fine.

If you use AGAT >= 1, you can make it in one command:
First you modify the config file to change output to GTF (`agat config --expose`) and then you can run the AGAT merge script (the sort is the same for all script output no need to re-run agat_convert_sp_gxf2gxf.pl).

**Reply by Dalhte** [reply]
Rebonjour
Je viens de voir que vous etiez francais, donc je me permets de changer de langue. Merci pour votre aide, ca marche bien. Comme je l'ai mis dans l'autre message, il me reste le probleme de la transformation de "transcript" en "RNA", mais le gft est suffisament petit pour le faire avec un crtl H (pas tres elelgant mais ca marche.
 Je ne veux pas trop vous soliciter mais j'ai un truc assez étrange en suite: J'ai essayé cellranger arc mkref sur mes deux gtf avant fusion est ca marche nikel (auutant le custum made que celui telechargé sur ensembl) par contre le merge ne marche pas. J'ai essayé un simple cat suivi agat_convert_sp_gxf2gxf.pl ainsi qu'un merge, ou un merge suivi d'un sorting, voir d'un agat_sp_Prokka_inferNameFromAttributes.pl en desepoir. J'obtiens systematiquement quelque chose du genre : 

Error while parsing GTF file /home/david/AGAT/RN6V104cat2.gtf
Error parsing GTF at line 567.  Parsed attribute had a quote in the middle of a value.  Please ensure quotes are only used to encapsulate attribute values.
 Bad Attribute Value = transcript_biotype
Ce n'est pas tout le temps la même ligne et le même biotype

Je ne comprend pas d'ou ca vient. 

J'ai essayer de faire une bidouille a blanc sur le GTF canonique (celui qui passe sans modif) en faisant un agat_sp_Prokka_inferNameFromAttributes.pl dessus et ca me genere le probleme. AGAT fait un truc etrange avec le GTF des bases de donnée. Pensez vous que c'est encore un probleme de format ?

Cordialement

David

**Reply by Dalhte** [reply]
Hello a nouveau.
J'ai trouvé le probleme je pense: AGAT rajoute une colonne tag avec ce genre d'info : tag "cds_start_NF" "mRNA_start_NF".  
J'ai tenté un
agat_sp_manage_attributes.pl --gff RattusV104Prokka.gtf -p tag --tag "cds_start_NF" "mRNA_start_NF" --output RattusV104Prokka2.gtf
Mais ca n'a pas réglé le probleme
j'ai tenté plus drastique :+1: (agat) david@Dalhte:~/AGAT$ agat_sp_manage_attributes.pl --gff RattusV104Prokka.gtf -p all  --tag tag --output RattusV104Prokka4.gtf 
Et la, cellranger-arc accepte de tourner :)
 Ma question: est ce que j'ai fait n'importe  quoi ? Y a t'il plus simple pour eviter que le merge contienne ces "tag" ?

Cordialement
David

**Comment by Juke34** [comment]
Normalement AGAT n'ajoute pas de tag tel que "cds_start_NF" "mRNA_start_NF". Ca devait être dans un des deux fichiers que vous avez mergé.
Si il y a des RNA qui apparaissent c'est que AGAT à detecter des features comme des exons qui n'ont pas de parents. Il me semble il crée un RNA car il ne voit pas de CDS à attacher... sinon je crois qu'il met mRNA.

Pour ce qui est de la dernière commande je ne comprends pas pourquoi le fichier de sortie fonctionne. Tout ce que vous avez fait c'est enlever les attributs "tag" de la 9em colonne, mais je pense pas qu'il est ce type d'attribut.
Je pense pas que cellranger à besoin de quelconque attribut à part ceux obligatoire pour lier les features entre-elles. Du coup l'approche standard serait:

Utilisez AGAT >=1.0.0
faire un `agat config --expose`, modifier l'option output pour avoir du GTF
lancez agat_sp_merge_annotations.pl pour merger vos fichiers
lancez  `agat_sp_manage_attributes.pl --gff in.gtf --tag all_attributes -o out.gtf` qui enlevera tous les attributs de la 9em colonnes sauf ceux obligatoires comme spécifié par le format.

**Comment by Dalhte** [comment]
Bonjour
Votre propostion a l'air de marcher parfaitement. Merci beaucoup pour votre aide très préciseuse !
Cordialement
David
